File: benchmarks/tsb/bench_to_latex.ts
/**
 * Benchmark: toLaTeX / seriesToLaTeX — render DataFrame/Series to LaTeX tabular format.
 *
 * Mirrors pandas:
 *   - `DataFrame.to_latex()` → tsb `toLaTeX(df)`
 *   - `Series.to_latex()`    → tsb `seriesToLaTeX(s)`
 *
 * Outputs JSON: {"function": "to_latex", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { DataFrame, Series, toLaTeX, seriesToLaTeX } from "../../src/index.ts";

const ROWS = 500;
const WARMUP = 5;
const ITERATIONS = 100;

const df = DataFrame.fromColumns({
  name: Array.from({ length: ROWS }, (_, i) => `item_${i}`),
  value: Float64Array.from({ length: ROWS }, (_, i) => i * 1.23),
  count: Float64Array.from({ length: ROWS }, (_, i) => i),
});

const s = new Series({ data: Float64Array.from({ length: ROWS }, (_, i) => i * 0.5) });

for (let i = 0; i < WARMUP; i++) {
  toLaTeX(df);
  toLaTeX(df, { index: false, booktabs: true });
  seriesToLaTeX(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  toLaTeX(df);
  toLaTeX(df, { index: false, booktabs: true });
  seriesToLaTeX(s);
}
const total_ms = performance.now() - start;

console.log(
  JSON.stringify({
    function: "to_latex",
    mean_ms: total_ms / ITERATIONS,
    iterations: ITERATIONS,
    total_ms,
  }),
);

File: benchmarks/pandas/bench_to_latex.py
"""Benchmark: toLaTeX / seriesToLaTeX — DataFrame.to_latex() and Series.to_latex() on 500 rows.

Mirrors tsb toLaTeX(df) / seriesToLaTeX(s) from src/stats/format_table.ts.
"""
import json
import time
import numpy as np
import pandas as pd

ROWS = 500
WARMUP = 5
ITERATIONS = 100

df = pd.DataFrame(
    {
        "name": [f"item_{i}" for i in range(ROWS)],
        "value": np.arange(ROWS) * 1.23,
        "count": np.arange(ROWS, dtype=float),
    }
)
s = pd.Series(np.arange(ROWS) * 0.5)

for _ in range(WARMUP):
    df.to_latex()
    df.to_latex(index=False)
    s.to_latex()

start = time.perf_counter()
for _ in range(ITERATIONS):
    df.to_latex()
    df.to_latex(index=False)
    s.to_latex()
total_ms = (time.perf_counter() - start) * 1000

print(
    json.dumps(
        {
            "function": "to_latex",
            "mean_ms": total_ms / ITERATIONS,
            "iterations": ITERATIONS,
            "total_ms": total_ms,
        }
    )
)
